### PR TITLE
Import code based on odml_torch from g3

### PR DESCRIPTION
**Pull request by majiddadashi**
BUG=357104656

### CODE REVIEW

File: ai_edge_torch/generative/examples/experimental/gemma/convert_to_tflite.py [hunk 1]
```diff
@@ -34,17 +34,17 @@ def convert_gemma_to_tflite(
...
       checkpoint_path (str): The filepath to the model checkpoint, or directory
         holding the checkpoint.
       prefill_seq_len (int, optional): The maximum size of prefill input tensor.
         Defaults to 512.
       kv_cache_max_len (int, optional): The maximum size of KV cache buffer,
         including both prefill and decode. Defaults to 1024.
-      quantize (bool, optional): Whether the model should be quanized.
-        Defaults to True.
+      quantize (bool, optional): Whether the model should be quanized. Defaults
+        to True.
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/experimental/gemma/convert_to_tflite.py:47]: the formatting seems a bit weird, why is it breaking to two lines?

File: ai_edge_torch/generative/examples/stable_diffusion/pipeline.py [hunk 2]
```diff
@@ -136,27 +138,36 @@ def run_tflite_pipeline(
...
     image quality.
   height:
     The height in pixels of the generated image.
   width:
     The width in pixels of the generated image.
   sampler:
-    A sampler to be used to denoise the encoded image latents. Can be one of `k_lms, `k_euler`,
+    A sampler to be used to denoise the encoded image latents. Can be one of
+    `k_lms, `k_euler`,
     or `k_euler_ancestral`.
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/pipeline.py:153]: again the formatting seems problematic here

File: ai_edge_torch/generative/test/test_model_conversion.py [hunk 3]
```diff
@@ -27,22 +24,24 @@
...
 
-class TestModelConversion(unittest.TestCase):
+
+class TestModelConversion(googletest.TestCase):
   """Unit tests that check for model conversion and correctness."""
 
   def test_toy_model_with_kv_cache(self):
     config = toy_model_with_kv_cache.get_model_config()
-    pytorch_model = toy_model_with_kv_cache.ToyModelWithKV(config)
+    pytorch_model = toy_model_with_kv_cache.ToyModelWithKV(config).eval()
```
Comment (haozha111) [inline ai_edge_torch/generative/test/test_model_conversion.py:35]: could you explain a bit why we need `.eval()`?

File: ai_edge_torch/generative/examples/experimental/gemma/convert_to_tflite.py (hunk 1 above)
Comment (majiddadashi) [reply ai_edge_torch/generative/examples/experimental/gemma/convert_to_tflite.py:47]: These formattings are done by PyFormat which is used for the code in g3.

As you can see on the left, it was already 2 lines before. This just moved "Defaults" one line up and had to move the rest down to fit under 80 characters.

File: ai_edge_torch/generative/examples/stable_diffusion/pipeline.py (hunk 2 above)
Comment (majiddadashi) [reply ai_edge_torch/generative/examples/stable_diffusion/pipeline.py:153]: The formatting looks right to me here. Mind clarifying?

File: ai_edge_torch/generative/test/test_model_conversion.py (hunk 3 above)
Comment (majiddadashi) [reply ai_edge_torch/generative/test/test_model_conversion.py:35]: This sets the module in evaluation (inference) mode.
https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.eval

`ai_edge_torch.convert` expects the module to be in this mode and emits a warning otherwise:
https://github.com/google-ai-edge/ai-edge-torch/tree/main/docs/pytorch_converter#providing-a-wrapper

It changes certain layers in the module preparing for inference. For example, for BatchNorm2d, it freezes the running statistics.

File: ai_edge_torch/_convert/fx_passes/test/test_build_aten_composite_pass.py [hunk 4]
```diff
@@ -12,20 +12,29 @@
...
+from ai_edge_torch._convert import fx_passes
 import torch
-import torch_xla
+
+from tensorflow.python.platform import googletest
 
 
 def _export_to_stablehlo_with_composite(
-    func: Union[torch.nn.Module, Callable], export_args
+    func: Union[torch.nn.Module, Callable[..., torch.Tensor]], export_args
```
Comment (chunnienc) [inline ai_edge_torch/_convert/fx_passes/test/test_build_aten_composite_pass.py:27]: Callable[..., Any]

**Review by chunnienc — DISMISSED** [review]
minor comment on typing, not a blocker for this to merge.

Checked all non-genai changes and looks good overall.

**Review by chunnienc — APPROVED** [review]
